### PR TITLE
[projects] Wrap ProjectDefinition in a class

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

Wraps some commonly needed project config values in one class. Also consolidates some helper functions.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.

### CODE REVIEW

File: python/ray/projects/projects.py [hunk 1]
```diff
@@ -2,12 +2,103 @@
...
+        if not os.path.exists(project_file):
+            raise ValueError("Project file {} not found".format(project_file))
+        with open(project_file) as f:
+            self.config = yaml.safe_load(f)
+
+        check_project_config(self.root, self.config)
+
+    def cluster_yaml(self):
+        """Return the project's cluster configuration filename.
+        """
```
Comment (richardliaw) [inline python/ray/projects/projects.py]: this should be same line as above 
`"""Return the project's cluster configuration filename."""

File: python/ray/projects/projects.py [hunk 2]
```diff
@@ -2,12 +2,103 @@
...
+        return self.config["cluster"]
+
+    def working_directory(self):
+        """Return the project's working directory on a cluster session.
+        """
+        directory = os.path.join("~", self.config["name"])
+        if not directory.endswith("/"):
```
Comment (richardliaw) [inline python/ray/projects/projects.py]: nit, you can do `os.path.join("~", self.config["name"], "")`, or generally `os.path.join(path, "")` for forcing path to be a directory

**Comment by pcmoritz** [comment]
Looks great, there is a few test failures from the refactor :)

**Review by pcmoritz — APPROVED** [review]
Great change, thanks :)

**Pull request merged**